apps/obsidian/src/utils/createNodeFromSelectedText.ts

```diff
@@ -0,0 +1,79 @@
+import { App, Editor, Notice, TFile } from "obsidian";
+import { DiscourseNode } from "~/types";
+import { getDiscourseNodeFormatExpression } from "./getDiscourseNodeFormatExpression";
+import { checkInvalidChars } from "./validateNodeType";
+
+export function formatNodeName(
+  selectedText: string,
+  nodeType: DiscourseNode,
+): string | null {
+  const regex = getDiscourseNodeFormatExpression(nodeType.format);
+  const nodeFormat = regex.source.match(/^\^(.*?)\(\.\*\?\)(.*?)\$$/);
+
+  if (!nodeFormat) return null;
+
+  return (
+    nodeFormat[1]?.replace(/\\/g, "") +
+    selectedText +
+    nodeFormat[2]?.replace(/\\/g, "")
+  );
+}
+
+export async function createDiscourseNodeFile(
+  app: App,
+  formattedNodeName: string,
+  nodeType: DiscourseNode,
+): Promise<TFile | null> {
+  try {
+    const existingFile = app.vault.getAbstractFileByPath(
+      `${formattedNodeName}.md`,
+    );
+    if (existingFile && existingFile instanceof TFile) {
+      new Notice(`File ${formattedNodeName} already exists`, 3000);
```

Comment:
We should create the link/reference, as this was the users intent.  Still good to notify them that the page wasn't created.